kafka/consumer-manager.go

```diff
@@ -0,0 +1,114 @@
+package kafka
+
+import (
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/Sirupsen/logrus"
+	"github.com/thrawn01/args"
+)
+
+type ConsumerManager struct {
+	parser    *args.ArgParser
+	done      chan struct{}
+	new       chan sarama.PartitionConsumer
+	reconnect chan bool
+	connected bool
+	mutex     sync.Mutex
+}
+
+func NewConsumerManager(parser *args.ArgParser) *ConsumerManager {
+	manager := &ConsumerManager{
+		parser: parser,
+	}
+	manager.Start()
+	return manager
+}
+
+func (self *ConsumerManager) connect() (sarama.PartitionConsumer, error) {
+	opts := self.parser.GetOpts()
+	logrus.Info("Connecting to Kafka Cluster ", opts.StringSlice("kafka-endpoints"))
+	consumer, err := sarama.NewConsumer(opts.StringSlice("kafka-endpoints"), nil)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"type":   "kafka",
+			"method": "NewConsumer()",
+		}).Error("Failed with - ", err.Error())
+		self.setConnected(false)
+		return nil, err
+	}
+	partition, err := consumer.ConsumePartition(opts.String("kafka-topic"), 0, sarama.OffsetNewest)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"type":   "kafka",
+			"method": "ConsumePartition()",
+		}).Error("Failed with - ", err.Error())
+		self.setConnected(false)
+		return nil, err
+	}
+	return partition, nil
+}
+
+func (self *ConsumerManager) GetConsumerChannel() chan sarama.PartitionConsumer {
```

Comment:
I'm in favor of them? LOL =D 

Originally I prefered ```/* comment */``` because of my c++ days, but the more I read the golang stdlib. The more I like the ```// comment ``` style.

I usually write ALOT of comments, but I've been told I write too many comments, so I've been holding back.